plugins/Signatures/class.signatures.plugin.php

```diff
@@ -267,7 +267,8 @@ public function CheckSignatureLength($Values, &$Sender) {
    public function CheckNumberOfImages($Values, &$Sender) {
       if (C('Plugins.Signatures.MaxNumberImages') && C('Plugins.Signatures.MaxNumberImages') !== 'Unlimited') {
          $max = C('Plugins.Signatures.MaxNumberImages');
-         $numMatches = preg_match_all('/(<img|\[img.*\]|\!\[.*\])/i', $Values['Plugin.Signatures.Sig']);
+         $Sig = Gdn_Format::To($Values['Plugin.Signatures.Sig'], $Sender->Form->GetFormValue('Format'));
```

Comment:
This is odd code. If the format isn't coming in `$Values` then it should. You also need to default the format to `C('Garden.InputFormatter')`.